### PR TITLE
feat(sentry): Add server_name option

### PR DESCRIPTION
**Pull request by loewenheim**
Unlike `default_tags`, the `server_name` also gets set for transactions. This might aid in the investigation of https://github.com/getsentry/team-ingest/issues/587.

### CODE REVIEW

**Review by jjbayer — APPROVED** [review]
Nice, don't forget to add a changelog entry.

File: relay-log/src/setup.rs [hunk 1]
```diff
@@ -173,6 +173,9 @@ pub struct SentryConfig {
     /// Sets the environment for this service.
     pub environment: Option<Cow<'static, str>>,
 
+    /// Sets the server name for this service.
+    pub server_name: Option<Cow<'static, str>>,
```
Comment (Dav1dde) [inline relay-log/src/setup.rs:180]: Can the SDK load the server name from an environment variable?

We should also make this env var overridable, especially for k8s deployments this is the only way to have a server name per pod.

File: relay-log/src/setup.rs [hunk 2]
```diff
@@ -287,12 +298,17 @@ pub fn init(config: &LogConfig, sentry: &SentryConfig) {
         .init();
 
     if let Some(dsn) = sentry.enabled_dsn() {
+        let server_name = env::var(SERVER_NAME_VAR)
+            .ok()
+            .map(Cow::from)
+            .or_else(|| sentry.server_name.clone());
```
Comment (loewenheim) [inline relay-log/src/setup.rs]: This prefers the environment variable if set, not sure if the other way around makes more sense.

Comment (Dav1dde) [reply relay-log/src/setup.rs]: We have [`apply_override`](https://github.com/getsentry/relay/blob/master/relay-config/src/config.rs#L1631) at the `Config` for env overrides, probably a better place to put it.

Comment (loewenheim) [reply relay-log/src/setup.rs]: Oh nice

File: relay/src/cli.rs [hunk 3]
```diff
@@ -119,6 +120,7 @@ pub fn extract_config_env_vars() -> OverridableConfig {
         outcome_source: None, //already extracted in params
         shutdown_timeout: env::var("SHUTDOWN_TIMEOUT").ok(),
         instance: env::var("RELAY_INSTANCE").ok(),
+        server_name: env::var("RELAY_SERVER_NAME").ok(),
```
Comment (Dav1dde) [inline relay/src/cli.rs]: We could default it to `HOSTNAME`.

Something like this (untested):
```rs
env::var("RELAY_SERVER_NAME").or_else(|| env::var("HOSTNAME")).ok()
```

**Review by Dav1dde — APPROVED** [review]

File: relay/src/cli.rs (hunk 3 above)
Comment (loewenheim) [reply relay/src/cli.rs]: I think you made a typo there

Comment (Dav1dde) [reply relay/src/cli.rs]: Updated, copy paste, but get the idea, check `RELAY_SERVER_NAME` then fallback to `HOSTNAME`

Comment (loewenheim) [reply relay/src/cli.rs]: Does the `HOSTNAME` variable have any special meaning/are we already using it somewhere?

Comment (Dav1dde) [reply relay/src/cli.rs]: A lot of (all?) kubernetes distributions set the `HOSTNAME` variable by default. Also e.g. [bash sets it](https://www.gnu.org/software/bash/manual/bash.html#Bash-Variables).

**Pull request merged**